stdpopsim/slim_engine.py

```diff
@@ -191,6 +233,12 @@
 
                     N0 = N[i,j];
                     r = Q * growth_rates[i,j];
+                    if (r < 0) {
+                        // N decreases, check size at end of "growth" phase.
+                        g_diff = growth_phase_end - growth_phase_start;
+                        N_growth_phase_end = asInteger(round(N0*exp(r*g_diff)));
+                        check_size(j, N_growth_phase_end, growth_phase_end);
```

Comment:
Wouldn't it help to factor out a function (or, maybe a lambda) that gives you the size of population `j` at time `t`, say? There's common code below.

Reply:
Hmm, yeah. I factored out the `epoch()` functionality, which is probably the wrong thing. I'll have a go at a pop_size_at() function.